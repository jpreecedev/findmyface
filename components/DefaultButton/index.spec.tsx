import React from "react"
import { render } from "@testing-library/react"
import "@testing-library/jest-dom/extend-expect"

import { DefaultButton } from "."

describe("<DefaultButton /> tests", () => {
  it("should render a default button", () => {
    const buttonTestId = "button"
    const buttonContentTestId = "button-content"

    const { getByTestId } = render(
      <DefaultButton>
        <span data-testid={buttonContentTestId}>Some button content</span>
      </DefaultButton>
    )

    expect(getByTestId(buttonTestId)).toMatchSnapshot()
    expect(getByTestId(buttonTestId)).toHaveClass("MuiButton-contained")
    expect(getByTestId(buttonTestId)).toHaveClass("MuiButton-containedPrimary")
    expect(getByTestId(buttonContentTestId)).toBeInTheDocument()
  })

  it("should render a large button", () => {
    const buttonTestId = "button"
    const buttonContentTestId = "button-content"

    const { getByTestId } = render(
      <DefaultButton size="large">
        <span data-testid={buttonContentTestId}>Some button content</span>
      </DefaultButton>
    )

    expect(getByTestId(buttonTestId)).toMatchSnapshot()
    expect(getByTestId(buttonTestId)).toHaveClass("MuiButton-contained")
    expect(getByTestId(buttonTestId)).toHaveClass("MuiButton-containedPrimary")
    expect(getByTestId(buttonTestId)).toHaveClass("MuiButton-sizeLarge")

    expect(getByTestId(buttonContentTestId)).toBeInTheDocument()
  })

  it("should render a secondary button", () => {
    const buttonTestId = "button"
    const buttonContentTestId = "button-content"

    const { getByTestId } = render(
      <DefaultButton color="secondary">
        <span data-testid={buttonContentTestId}>Some button content</span>
      </DefaultButton>
    )

    expect(getByTestId(buttonTestId)).toMatchSnapshot()
    expect(getByTestId(buttonTestId)).toHaveClass("MuiButton-contained")
    expect(getByTestId(buttonTestId)).toHaveClass("MuiButton-containedSecondary")

    expect(getByTestId(buttonContentTestId)).toBeInTheDocument()
  })

  it("should render an outlined secondary button", () => {
    const buttonTestId = "button"
    const buttonContentTestId = "button-content"

    const { getByTestId } = render(
      <DefaultButton color="secondary" variant="outlined">
        <span data-testid={buttonContentTestId}>Some button content</span>
      </DefaultButton>
    )

    expect(getByTestId(buttonTestId)).toMatchSnapshot()
    expect(getByTestId(buttonTestId)).toHaveClass("MuiButton-outlined")
    expect(getByTestId(buttonTestId)).toHaveClass("MuiButton-outlinedSecondary")

    expect(getByTestId(buttonContentTestId)).toBeInTheDocument()
  })
})
