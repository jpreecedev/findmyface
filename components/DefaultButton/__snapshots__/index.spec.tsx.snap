// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DefaultButton /> tests should render a default button 1`] = `
<button
  class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-button-1 MuiButton-containedPrimary"
  data-testid="button"
  tabindex="0"
  type="button"
>
  <span
    class="MuiButton-label"
  >
    <span
      data-testid="button-content"
    >
      Some button content
    </span>
  </span>
  <span
    class="MuiTouchRipple-root"
  />
</button>
`;

exports[`<DefaultButton /> tests should render a large button 1`] = `
<button
  class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-button-43 MuiButton-containedPrimary MuiButton-containedSizeLarge MuiButton-sizeLarge"
  data-testid="button"
  tabindex="0"
  type="button"
>
  <span
    class="MuiButton-label"
  >
    <span
      data-testid="button-content"
    >
      Some button content
    </span>
  </span>
  <span
    class="MuiTouchRipple-root"
  />
</button>
`;

exports[`<DefaultButton /> tests should render a secondary button 1`] = `
<button
  class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-button-85 MuiButton-containedSecondary"
  data-testid="button"
  tabindex="0"
  type="button"
>
  <span
    class="MuiButton-label"
  >
    <span
      data-testid="button-content"
    >
      Some button content
    </span>
  </span>
  <span
    class="MuiTouchRipple-root"
  />
</button>
`;

exports[`<DefaultButton /> tests should render an outlined secondary button 1`] = `
<button
  class="MuiButtonBase-root MuiButton-root MuiButton-outlined makeStyles-button-127 MuiButton-outlinedSecondary"
  data-testid="button"
  tabindex="0"
  type="button"
>
  <span
    class="MuiButton-label"
  >
    <span
      data-testid="button-content"
    >
      Some button content
    </span>
  </span>
  <span
    class="MuiTouchRipple-root"
  />
</button>
`;
